packages/engine/src/simulation/package/state/packages/behavior_execution/mod.rs

```diff
@@ -239,7 +236,6 @@ impl Package for BehaviorExecution {
         self.fix_behavior_chains(agent_pool)?;
         self.reset_behavior_index_col(agent_pool)?;
         agent_pool.flush_pending_columns()?;
-        drop(agent_pool);
```

Comment:
I think I agree with removing this. It just feels like it makes sense to release access when you're done needing it, but we're trying to have guarantees about access anyway and avoid deadlocks so I don't think it actually gives us a benefit as something shouldn't be waiting on it, right?

Reply:
`agent_pool` is a reference and clippy forbids `drop(reference)` (for good reasons).

Reply:
We could perhaps look at just doing that code within a scoped block `{ ... }` but again not sure it brings any value